tests/test_smoke.py

```diff
@@ -2646,6 +2704,66 @@ def test_invalid_names(self, invalid_name_list, store_type):
                 storage_obj = storage_lib.Storage(name=name)
                 storage_obj.add_store(store_type)
 
+    @pytest.mark.parametrize('ext_bucket_fixture, store_type',
+                             [('tmp_awscli_bucket', storage_lib.StoreType.S3),
+                              ('tmp_gsutil_bucket', storage_lib.StoreType.GCS),
+                              pytest.param('tmp_awscli_bucket_r2',
+                                           storage_lib.StoreType.R2,
+                                           marks=pytest.mark.cloudflare)])
+    def test_upload_to_existing_bucket(self, ext_bucket_fixture, request,
+                                       tmp_source, store_type):
+        # Tries uploading existing files to newly created bucket (outside of
+        # sky) and verifies that files are written.
+        bucket_name = request.getfixturevalue(ext_bucket_fixture)
+        storage_obj = storage_lib.Storage(name=bucket_name, source=tmp_source)
+        storage_obj.add_store(store_type)
+
+        # Check if tmp_source/tmp-file exists in the bucket using aws cli
+        out = subprocess.check_output(self.cli_ls_cmd(store_type, bucket_name),
+                                      shell=True)
+        assert 'tmp-file' in out.decode('utf-8'), \
+            'File not found in bucket - output was : {}'.format(out.decode
+                                                                ('utf-8'))
+
+    @pytest.mark.parametrize('ext_bucket_fixture, store_type',
+                             [('tmp_awscli_bucket', storage_lib.StoreType.S3),
+                              ('tmp_gsutil_bucket', storage_lib.StoreType.GCS),
```

Comment:
We should probably separate these fixtures from those used in test_upload_to_existing_bucket. Is it possible to add an arg to these fixtures to set the skymanaged label?

Reply:
Currently, it's only the test `test_storage_refresh` requiring the tagging, so I added a line in fixture to check if the calling test name is `test_storage_refresh` to set the flag.